### ISSUE TITLE
The output of the train and eval

### ISSUE THREAD

**Issue by PuJanhan** [WeijingShi/Point-GNN#65]
I  use   python3 train.py configs/car_auto_T3_train_train_config configs/car_auto_T3_train_config   and  python3 eval.py configs/car_auto_T3_train_eval_config,    they  get the results like this :
Class_0: recall=0.992798, prec=0.992376, mAP=0.999531, loc=0.000000
         x=0.0000 y=0.0000 z=0.0000 l=0.0000 h=0.0000 w=0.0000 y=0.0000
Class_1: recall=0.807884, prec=0.801547, mAP=0.872123, loc=7.071270
         x=0.7826 y=0.5973 z=2.0156 l=0.9511 h=0.5748 w=0.6402 y=1.5098
Class_2: recall=0.838907, prec=0.830165, mAP=0.911929, loc=19.158129
         x=0.9189 y=1.2436 z=8.2396 l=2.5387 h=1.5434 w=1.7170 y=2.9570
Class_3: recall=0.783943, prec=0.834315, mAP=0.873242, loc=0.000000
         x=0.0000 y=0.0000 z=0.0000 l=0.0000 h=0.0000 w=0.0000 y=0.0000
but in your paper ,three types of car such as easy moderate hard , they has their own map, i dont konw where get the results of paper in your code and the dataset dont have the three types . In addition,what the correspondence between the output of train and the results in your paper. which result in the code represent the result in your paper

**Comment by WeijingShi** [comment]
Hi @PuJanhan, Thanks for the interest. 

The printout from the scripts are at the point level. E.g. the recall of car points gets detected (thres=0.5).  To get the object level mAP, use  submodule kitti_native_evaluation:

Install kitti_native_evaluation offline evaluation:
```
cd kitti_native_evaluation
cmake ./
make
```
Evaluate output results on the validation split:
```
evaluate_object_offline DATASET_ROOT_DIR/labels/training/label_2/ DIR_TO_SAVE_RESULTS
```

It will give the easy moderate hard category as well.

**Comment by typhoonlee** [comment]
> I use python3 train.py configs/car_auto_T3_train_train_config configs/car_auto_T3_train_config and python3 eval.py configs/car_auto_T3_train_eval_config, they get the results like this :
> Class_0: recall=0.992798, prec=0.992376, mAP=0.999531, loc=0.000000
> x=0.0000 y=0.0000 z=0.0000 l=0.0000 h=0.0000 w=0.0000 y=0.0000
> Class_1: recall=0.807884, prec=0.801547, mAP=0.872123, loc=7.071270
> x=0.7826 y=0.5973 z=2.0156 l=0.9511 h=0.5748 w=0.6402 y=1.5098
> Class_2: recall=0.838907, prec=0.830165, mAP=0.911929, loc=19.158129
> x=0.9189 y=1.2436 z=8.2396 l=2.5387 h=1.5434 w=1.7170 y=2.9570
> Class_3: recall=0.783943, prec=0.834315, mAP=0.873242, loc=0.000000
> x=0.0000 y=0.0000 z=0.0000 l=0.0000 h=0.0000 w=0.0000 y=0.0000
> but in your paper ,three types of car such as easy moderate hard , they has their own map, i dont konw where get the results of paper in your code and the dataset dont have the three types . In addition,what the correspondence between the output of train and the results in your paper. which result in the code represent the result in your paper

Hi，Did you verify the model successfully?

**Comment by PuJanhan** [comment]
Sorry,I don't have time and equipment to do it recently. I will try it later. thanks for your reply------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;&quot;typhoonlee&quot;<notifications@github.com&gt;
发送时间:&nbsp;2021年1月18日(星期一) 上午10:10
收件人:&nbsp;&quot;WeijingShi/Point-GNN&quot;<Point-GNN@noreply.github.com&gt;;
抄送:&nbsp;&quot;PuJH&quot;<805280381@qq.com&gt;;&quot;Mention&quot;<mention@noreply.github.com&gt;;
主题:&nbsp;Re: [WeijingShi/Point-GNN] The output of the train and eval (#65)

**Comment by PuJanhan** [comment]
Yes ,I follow the advice of [https://github.com/WeijingShi/Point-GNN/issues/69](url)and success
